### PR TITLE
Add double quotes around col names to handle reserved keywords

### PR DESCRIPTION
**Pull request by ravjotbrar**
### Summary

This change addresses how reserved keywords used as column names are handled by our connector.

### Description

Added double quotes around column names in reads, writes, and merges to be able to handle Vertica's reserved keywords. Also added integration tests to test this.  

### Related Issue

https://github.com/vertica/spark-connector/issues/149

### Additional Reviewers

@alexey-temnikov 
@alexr-bq 
@jonathanl-bq

### CODE REVIEW

File: connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala [hunk 1]
```diff
@@ -328,10 +328,10 @@ class SchemaTools extends SchemaToolsInterface {
             typenameNormalized.startsWith("uuid")) {
             castToVarchar(info.label)
           } else {
-            info.label
+            "\"" + info.label + "\""
```
Comment (alexr-bq) [inline connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala:331]: I think we should support the user specifying the column names with quotes around them, and check for that.

**Review by alexr-bq — CHANGES_REQUESTED** [review]

File: connector/src/main/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemWritePipe.scala [hunk 2]
```diff
@@ -164,7 +164,9 @@ class VerticaDistributedFilesystemWritePipe(val config: DistributedFilesystemWri
       case Left(err) => Left(MergeColumnListError(err))
     }
     val mergeList = config.mergeKey match {
-      case Some(keys) => keys.toString.split(",").toList.map(col => s"target.$col=temp.$col").mkString(" AND ")
+      case Some(key) =>
+        val trimmedCols= key.toString.split(",").toList.map(col => col.trim())
```
Comment (jonathanl-bq) [inline connector/src/main/scala/com/vertica/spark/datasource/core/VerticaDistributedFilesystemWritePipe.scala:168]: nit: space before the = sign

File: connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala [hunk 3]
```diff
@@ -311,7 +311,7 @@ class SchemaTools extends SchemaToolsInterface {
     } yield columnList
   }
 
-  private def castToVarchar: String => String = colName => colName + "::varchar AS " + colName
+  private def castToVarchar: String => String = colName => colName + "::varchar AS " + "\"" + colName + "\""
```
Comment (jonathanl-bq) [inline connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala:314]: Might be worth making a helper function to just add escaped quotes around strings.

**Review by jonathanl-bq — APPROVED** [review]

File: connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala (hunk 1 above)
Comment (ravjotbrar) [reply connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala:331]: done

File: connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala (hunk 3 above)
Comment (ravjotbrar) [reply connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala:314]: done

**Review by alexr-bq — APPROVED** [review]

**Pull request merged**